src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -236,6 +248,27 @@ public HttpResponse doAbort() {
         return HttpResponses.ok();
     }
 
+    private void notifyInput(InputExtension.NotifyEvent notifyEvent) {
+        InputStep inputStep = this.getInput();
+        Jenkins.get().getExtensionList(InputExtension.class).forEach(input -> {
+            final String extensionName = input.getName();
+
+            User currentUser = User.current();
+            String userID = "anonymous";
+            if(currentUser != null) {
+                userID = currentUser.getId();
+            }
+
+            try {
+                input.notifyInput(inputStep, run, userID, notifyEvent);
+
+                listener.getLogger().printf("Notification for %s succeed.", extensionName);
```

Comment:
I'd remove this logging and let the extensions log if desired. I'd also delete `InputExtension.getName` and use the class name directly if an exception is caught.